src/main/java/CommandParser.java

```diff
@@ -0,0 +1,80 @@
+public class CommandParser {
+    private Commands flag;
+    private String parameter;
+
+    public CommandParser(){
+        flag = Commands.UNKNOWN;
+        parameter = Constants.ZERO_LENGTH_STRING;
+    }
+
+    public void parse(String rawInput){
+        if (rawInput == null) return;
+        String[] input = rawInput.split(Constants.SPACE);
+        updateFlag(input[0]);
+        if (flag == Commands.DEADLINE || flag == Commands.EVENT ||
+                flag == Commands.DONE || flag == Commands.UNDONE){
+            String commandSign = getCommandSign(flag);
+            if(rawInput.contains(commandSign)){
+                parameter = rawInput.split(commandSign)[1].trim();
+            }
+            else {
+                flag = Commands.ILLEGAL;
+            }
+        }
+        else {
+            parameter = Constants.ZERO_LENGTH_STRING;
+        }
+    }
+
+    public Commands getFlag(){
+        return flag;
+    }
+
+    public String getParameter(){
+        return parameter;
+    }
+
+    private String getCommandSign(Commands command){
+        switch (command){
+        case DEADLINE:
+            return Constants.DDL_SIGN;
+        case EVENT:
+            return Constants.EVENT_SIGN;
+        case DONE:
+            return Constants.DONE_CMD;
+        case UNDONE:
+            return Constants.UNDONE_CMD;
+        default:
+            return Constants.ZERO_LENGTH_STRING;
+        }
+    }
```

Comment:
Consider adding "break;" statements unless it is intended

Reply:
Thanks bro 👍 Just that the IDE will underline all "break"s after "return"s because they are unreachable code, so I removed them.